### DISCUSSION TITLE
`BASE_URL` ignores path component

### DISCUSSION THREAD

**Discussion by nh2** [plausible/analytics#1184] [Self-Hosted Support] [answered]
## Bug report

I'm trying to set `BASE_URL=https://example.com/plausible`, so that the root of Plausible starts at `/plausible`.

However, it does not work: Plausible redirects to `https://example.com/login`, outside of the `BASE_URL`.

I believe this is because Plausible completely ignores the `.path` component of the parsed URI:

https://github.com/plausible/analytics/blob/145a3599c1c5f7e7c9c1fccc34be311f88dd5e06/config/runtime.exs#L181

**Expected behavior**

Plausible should respect the path component of the `BASE_URL`, so that it can run under a path that is not the domain root.

In https://github.com/plausible/analytics/discussions/177 it was mentioned that Plausible strips query parameters after `?`, but nowhere is it mentioned that it strips the `/path`, so I assume that this is accidental.

**Comment by ukutaht** [comment, marked as answer]
Yes this does seem like a problem that stops Plausible from being used in a subdirectory as it stands. I think the fix is as easy as adding `[path: base_url.path]` to the config line but I'd have to test it.

**Reply by nuno-agostinho** [reply]
Hey @ukutaht, any news on this? I would love to set a custom root for plausible. Thanks! :)

**Reply by ukutaht** [reply]
No news but I've added it to my list for the next release. Expected in October or November

**Reply by daguitosama** [reply]
I i think it will be terrific to mention this in the getting started docs, it can save tons of time to know that when you get a 404 on BASE_URL and 500 on anything else, it's probable that you have used a `path` component in that URL, not anything else, you don't need to go mental with docker/proxy-server/os-platform-integration-issues theories of what is happening. And trust me, this issue can take time to be found 😅️

**Comment by ukutaht** [comment]
Hi, thanks for reporting. This should be fixed in https://github.com/plausible/analytics/releases/tag/v1.4.0-rc.0. Can someone test and confirm?

**Reply by nuno-agostinho** [reply]
Hey @ukutaht, I tried to set the root of Plausible at `/plausible` using [Plausible 1.4.0](https://github.com/plausible/analytics/releases/tag/v1.4.0). However, it doesn't seem to be working for me.

When setting the root to a path, it seems that some resources return 404. If I remove the `/plausible` from the web address, they exist. For instance, one of the resources that it tries to load is `http://localhost:8000/plausible/images/icon/plausible_logo_dark-9fff186187e2a9c972ae81c88e63e830.png` with no success. But the resource is indeed available at `http://localhost:8000/images/icon/plausible_logo_dark-9fff186187e2a9c972ae81c88e63e830.png?vsn=d`.

<img width="1081" alt="image" src="https://user-images.githubusercontent.com/3199157/141314361-0f41199d-ad87-416b-a2c7-2fea0aa70ad3.png">

Do you prefer if I open a GitHub issue on this?

**Reply by daguitosama** [reply]
I have expirienced a similar issue a few hours ago with a BASE_URL as `https://my-domain/analytics` i get :
`
/analytics                                 ->    404
`
`
{anithing-else\.(?:js|css)}    ->    500
`

![Screenshot from 2021-11-11 04-30-37 edited](https://user-images.githubusercontent.com/34744883/141345963-1f3d26de-4e45-4047-9736-db37b5c12359.png)

**Reply by ukutaht** [reply]
So I tried fixing BASE_URL for the 1.4.0 release but it turned out to be a much bigger project than I anticipated. So officially subfolder installations are not supported yet, sorry.

**Reply by nuno-agostinho** [reply]
Thanks @ukutaht, I know that feeling when a feature is much harder to implement than planned... Thank you for trying! 😄

**Reply by daguitosama** [reply]
Don't worry @ukutaht i appreciate you make the time to look it up, work it around is supper easy, just make a sub domain for it and you are all set, but I highly recommend give this heads up in the selfhosted/getting-started doc: 
" [important] 
Hey yooos do NOT USE a path based url for the BSAE_URL ok "
I can save a lot of time until people find this issue 🤣

**Reply by ukutaht** [reply]
Thanks @d4g0! I've added some notes here: https://github.com/plausible/docs/commit/6f5780c07c75f61d2f121c361952c56d308eac82

Feel free to open a PR if you have a better way to explain this limitation

**Reply by LuisSchweinberger** [reply]
Hi @ukutaht,

is this feature still planned to be released. We want to use plausible for a uni project and on our server we are not allowed to create subdomains. If Not is there maybe a workaround you can recommend?

Thanks in advance

**Reply by ruslandoga** [reply]
@LuckyLuisHD 👋

Here's one workaround I saw recently: https://github.com/plausible/analytics/discussions/3561#discussioncomment-7684820

**Comment by dandaniel5** [comment]
@ukutaht  what exectly should i add to what config ? .env or compose.yml ?

**Reply by ruslandoga** [reply]
👋 @dandaniel5 

Setting path components on BASE_URL is not supported at the moment.

From the discussion above:

> Plausible is not currently designed for subfolder installations so please do not add a path component to the base url.

**Reply by dandaniel5** [reply]
@ukutaht pls tell me how am I supposed to use it if it can be hosted in the main domain directory?
Do I need to add it to my main server with a separate URL?
nginx

**Reply by ruslandoga** [reply]
> Do I need to add it to my main server with a separate URL?

Yes. Please see https://github.com/plausible/community-edition/wiki/reverse-proxy for more information :)

**Reply by dandaniel5** [reply]
@ukutaht  
it says use BASE_URL...
my NGINX is fine. but plausible use wrong redirects igoring BASE_URL, so ?

**Reply by ruslandoga** [reply]
So you must have configured it wrong :)

**Reply by dandaniel5** [reply]
@ukutaht
I need to use it on my main domain at a specific address, something like /plausible. I’ve set everything up correctly, including adding the WebSocket, /js/script.js, and /api/event.
But when Plausible does an internal redirect from page to page, it uses the wrong base URL inside.
pls ASAP someone qualified
